src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
+    [button addTarget:self
+               action:@selector(showFlutter)
+     forControlEvents:UIControlEventTouchUpInside];
+    [button setTitle:@"Show Flutter!" forState:UIControlStateNormal];
+    [button setBackgroundColor:[UIColor blueColor]];
+    button.frame = CGRectMake(80.0, 210.0, 160.0, 40.0);
+    [self.view addSubview:button];
+}
+
+- (void)showFlutter {
+    FlutterEngine *flutterEngine =
+        [(AppDelegate *)[[UIApplication sharedApplication] delegate] flutterEngine];
+    FlutterViewController *flutterViewController =
+        [[FlutterViewController alloc] initWithEngine:flutterEngine nibName:nil bundle:nil];
+    [self presentViewController:flutterViewController animated:YES completion:nil];
 }
+@end
 ```
 
-<details>
-<summary>What to do if the app delegate already inherits from somewhere else.</summary>
+</div>
+
+<div class="tab-pane" id="vc-swift-tab" role="tabpanel" aria-labelledby="vc-swift-tab" markdown="1">
+
+<?code-excerpt "ViewController.swift" title?>
+```swift
+import UIKit
+import Flutter
+
+class ViewController: UIViewController {
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    // Make a button to call the showFlutter function below when pressed.
+    let button = UIButton(type:UIButton.ButtonType.custom)
+    button.addTarget(self, action: #selector(showFlutter), for: .touchUpInside)
+    button.setTitle("Show Flutter!", for: UIControl.State.normal)
+    button.frame = CGRect(x: 80.0, y: 210.0, width: 160.0, height: 40.0)
+    button.backgroundColor = UIColor.blue
+    self.view.addSubview(button)
+  }
+
+  @objc func showFlutter() {
+    let flutterEngine = (UIApplication.shared.delegate as! AppDelegate).flutterEngine
+    let flutterViewController =
+        FlutterViewController(engine: flutterEngine, nibName: nil, bundle: nil)
+    present(flutterViewController, animated: true, completion: nil)
+  }
+}
+```
+
+</div>
+
+</div>{% comment %} FlutterViewController language tab end {% endcomment -%}
+
+Now you have a Flutter screen embedded in your iOS app.
+
+{{site.alert.note}}
+Using the example above, the default `main()` entrypoint function of your
+default Dart library would have been run when calling `run` on the
+`FlutterEngine` created in the `AppDelegate`.
+{{site.alert.end}}
+
+### _Alternatively_ - Create a FlutterViewController with an implicit FlutterEngine
+
+As an alternative to the above example, you can also let the
+`FlutterViewController` implicitly create its own `FlutterEngine` without
+pre-warming one ahead of time.
 
-Make your app delegate implement the `FlutterAppLifeCycleProvider` protocol, for
-example:
+This is not recommended since creating a `FlutterEngine` on-demand could
+introduce a noticeable latency between when the `FlutterViewController` is
+presented and when it will render its first frame. This could, however, be
+sometimes useful if the Flutter screen is rarely shown and there are no good
+heuristics to determine when the Dart VM should be started.
```

Comment:
`if the Flutter screen is rarely shown and does not need to persist state between apperances`?

Reply:
added into an enumeration